src/doc/TextDocument.ts

```diff
@@ -270,26 +295,13 @@ function getAttributes(Type: any, data: any, from: number, to: number, filter?:
     index += Type.length(next);
     if (index > from && (!filter || filter(next))) {
       if (!next.attributes) attributes = {};
-      else if (!attributes) {
-        attributes = { ...next.attributes };
-        if (attributes) delete attributes.id;
-      } else attributes = intersectAttributes(attributes, next.attributes);
+      else if (!attributes) attributes = { ...next.attributes };
+      else attributes = intersectAttributes(attributes, next.attributes);
     }
   }
   return attributes || EMPTY_OBJ;
 }
 
-function applyDeltaToLines(delta: Delta, lines: Line[], byId: LineIds) {
-  if (!lines.length) return lines;
-  const applied = Line.toDelta(lines, true).compose(delta, true);
-  while (applied.ops.length && !applied.ops[applied.ops.length - 1].insert) applied.ops.pop();
-  return Line.fromDelta(applied, byId).map(line => {
-    const id = Line.getId(line);
-    const old = byId.get(id);
-    return old && Line.equal(old, line) ? old : line;
-  });
-}
-
```

Comment:
The old method of `Line` > `Delta` > `compose()` > `Line` that prevented us from handling the line ids the way we wanted and I'm sure was less efficient.